### PR TITLE
Bundle improvements

### PR DESCRIPTION
**Pull request by vinicius73**
This PR follows changes made in https://github.com/nextcloud/nextcloud-event-bus/pull/481

- Update README.md
  - Install instructions
  - Usase instructions
  - Doc references 
- Update dependencies
- Use rollup as bundler
  - Generate commonjs and ESM bundle compatible.
- Remove babel
  - Typescript already is able to generate code that is in compliance with es5
- Remove unnecessary files from the npm package

### CODE REVIEW

File: lib/user.ts [hunk 1]
```diff
@@ -1,6 +1,11 @@
 /// <reference types="@nextcloud/typings" />
 
-declare var OC: Nextcloud.v16.OC | Nextcloud.v17.OC | Nextcloud.v18.OC | Nextcloud.v19.OC | Nextcloud.v20.OC;
+declare var OC: Nextcloud.v16.OC
+	| Nextcloud.v17.OC
+	| Nextcloud.v18.OC
+	| Nextcloud.v19.OC
+	| Nextcloud.v20.OC
+	| Nextcloud.v24.OC;
```
Comment (ChristophWurst) [inline lib/user.ts:8]: Also add 21 to 23?

File: package.json [hunk 2]
```diff
@@ -1,14 +1,22 @@
 {
   "name": "@nextcloud/auth",
-  "version": "1.3.0",
-  "description": "",
+  "version": "2.0.0-next",
+  "description": "Nextcloud helpers related to authentication and the current user",
```
Comment (ChristophWurst) [inline package.json:4]: I would say it is authentication and authorization (*isAdmin*)

Comment (vinicius73) [reply package.json:4]: I've just copied from package repo 🤣

File: lib/user.ts (hunk 1 above)
Comment (vinicius73) [reply lib/user.ts:8]: Sure, why not?

**Review by artonge — APPROVED** [review]

**Comment by nickvergessen** [comment]
Sorry, didn't know about this and pushed https://github.com/nextcloud/nextcloud-auth/pull/454

What is actually needed to get this in?

**Comment by vinicius73** [comment]
> What is actually needed to get this in?

I'm waiting for new release from `nextcloud-event-bus`

https://github.com/nextcloud/nextcloud-event-bus/pull/490

**Comment by vinicius73** [comment]
Now `@nextcloud/auth` v2 is ~60% smaller than v1

https://bundlephobia.com/package/@nextcloud/auth

Ref: https://github.com/nextcloud/nextcloud-axios/issues/151

**Pull request merged**